python/tests/integration/toolbox/test_library_tool.py

```diff
@@ -137,3 +138,83 @@ def test_count_keys(object_and_mem_and_lmdb_version_store):
     assert lib_tool.count_keys(KeyType.SNAPSHOT_REF) == 1
     assert lib_tool.count_keys(KeyType.MULTI_KEY) == 1
     assert lib_tool.count_keys(KeyType.SNAPSHOT) == 0
+
+
+@pytest.mark.parametrize("use_time_index", [True, False])
+def test_read_data_key_from_version_ref(object_and_mem_and_lmdb_version_store, use_time_index):
+    lib = object_and_mem_and_lmdb_version_store
+    lib_tool = lib.library_tool()
+    sym = "sym"
+
+    df = pd.DataFrame({"a": [1, 2, 3]})
+    if use_time_index:
+        df = pd.DataFrame(index=pd.date_range(start=pd.Timestamp(0), periods=3), data={"a": [1, 2, 3]})
+    lib.write(sym, df)
+
+    ver_ref_key = lib_tool.find_keys_for_symbol(KeyType.VERSION_REF, sym)[0]
+    ver_key = lib_tool.read_to_keys(ver_ref_key)[-1]
+    index_key = lib_tool.read_to_keys(ver_key)[0]
+    data_key = lib_tool.read_to_keys(index_key)[0]
```

Comment:
Assertions on the structure of these 3 keys, and checks that they can be used to read segments